### PR TITLE
8283626: AArch64: Set relocInfo::offset_unit to 4

### PR DESCRIPTION
**Pull request by eastig**
AArch64 instructions size is 4 bytes. Two lowest bits of `relocInfo` offsets can be discarded.
Tested with release and fastdebug builds:
- `gtest`: Passed.
- `tier1`...`tier4`: Passed

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8283626](https://bugs.openjdk.java.net/browse/JDK-8283626): AArch64: Set relocInfo::offset_unit to 4

### Reviewers
 * [Volker Simonis](https://openjdk.java.net/census#simonis) (@simonis - **Reviewer**)
 * [Andrew Haley](https://openjdk.java.net/census#aph) (@theRealAph - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7991/head:pull/7991` \
`$ git checkout pull/7991`

Update a local copy of the PR: \
`$ git checkout pull/7991` \
`$ git pull https://git.openjdk.java.net/jdk pull/7991/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7991`

View PR using the GUI difftool: \
`$ git pr show -t 7991`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7991.diff">https://git.openjdk.java.net/jdk/pull/7991.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/cpu/aarch64/relocInfo_aarch64.hpp [hunk 1]
```diff
@@ -29,8 +29,9 @@
   // machine-dependent parts of class relocInfo
  private:
   enum {
-    // Relocations are byte-aligned.
-    offset_unit        =  1,
+    // AArch64 instructions size is 4 bytes.
+    // Two lowest offset bits can always be discarded.
```
Comment (theRealAph) [inline src/hotspot/cpu/aarch64/relocInfo_aarch64.hpp:33]:
```suggestion
    // AArch64 instructions are always 4 bytes long and 4-aligned, so the
    // two lowest offset bits can always be discarded.
```

**Comment by theRealAph** [comment]
Looks good. I remember this being suggested a long time ago, but we were close to release and I was worried something might break unexpectedly.

File: src/hotspot/cpu/aarch64/relocInfo_aarch64.hpp (hunk 1 above)
Comment (eastig) [reply src/hotspot/cpu/aarch64/relocInfo_aarch64.hpp]: @theRealAph Thank you for reviewing and comments.

**Review by simonis — APPROVED** [review]
Looks good to me.

This also fixes the issue that currently `OptoLoopAlignment` can be set to `1` although it should be a *multiple of NOP size* according to `OptoLoopAlignmentConstraintFunc()`.

**Review by theRealAph — APPROVED** [review]

**Pull request closed without merging**